src/index.js

```diff
@@ -143,6 +141,51 @@ class SWPrecacheWebpackPlugin {
     }
   }
 
+  configureImportScripts(importScripts, publicPath, compiler, compilation) {
+    if (!importScripts) {
+      return;
+    }
+
+    const {hash, chunks} = compilation.getStats()
+      .toJson({hash: true, chunks: true});
+
+    this.overrides.importScripts = importScripts
+      .reduce((fileList, criteria) => {
+        // legacy support for importScripts items defined as string
+        if (typeof criteria === 'string') {
+          criteria = {filename: criteria};
+        }
+
+        const hasFileName = !!criteria.filename;
+        const hasChunkName = !!criteria.chunkName;
+
+        if (hasFileName && hasChunkName) {
+          this.warnings.push(new Error(
+            format(CHUNK_NAME_OVERRIDES_FILENAME_WARNING, criteria)
+          ));
+        }
+
+        if (hasChunkName) {
+          const chunk = chunks.find(c => c.names.includes(criteria.chunkName));
+
+          if (!chunk) {
+            compilation.errors.push(new Error(
```

Comment:
I think this should report an error, because it can only happen if they specify a chunk name which does not exist.